src/discussions/learners/data/slices.js

```diff
@@ -0,0 +1,98 @@
+/* eslint-disable no-param-reassign,import/prefer-default-export */
+import { createSlice } from '@reduxjs/toolkit';
+
+import {
+  AllPostsFilter,
+  MyPostsFilter,
+  PostsStatusFilter,
+  RequestStatus,
+  ThreadOrdering,
+} from '../../../data/constants';
+
+const learnersSlice = createSlice({
+  name: 'learner',
+  initialState: {
+    status: RequestStatus.IN_PROGRESS,
+    pages: [],
+    nextPage: null,
+    totalPages: null,
+    totalLearners: null,
+    postStatus: RequestStatus.SUCCESSFUL,
+    filters: {
+      status: PostsStatusFilter.ALL,
+      allPosts: AllPostsFilter.ALL_POSTS,
+      myPosts: MyPostsFilter.MY_POSTS,
+      search: '',
+    },
+    sortedBy: ThreadOrdering.BY_LAST_ACTIVITY,
+  },
+  reducers: {
+    fetchLearnersRequest: (state) => {
+      state.status = RequestStatus.IN_PROGRESS;
+    },
+    fetchLearnersSuccess: (state, { payload }) => {
+      state.status = RequestStatus.SUCCESSFUL;
+      state.learners = payload.results;
+      state.learnersProfile = payload.learnersProfile || [];
+      state.nextPage = payload.pagination.next;
+      state.totalPages = payload.pagination.num_pages;
+      state.totalLearners = payload.pagination.count;
+    },
+    fetchLearnersFailed: (state) => {
+      state.status = RequestStatus.FAILED;
+    },
+    fetchLearnersDenied: (state) => {
+      state.status = RequestStatus.DENIED;
+    },
+    fetchThreadRequest: (state) => {
+      state.status = RequestStatus.IN_PROGRESS;
+    },
+    fetchLearnersuccess: (state, { payload }) => {
+      state.status = RequestStatus.SUCCESSFUL;
+      state.LearnersById = { ...state.LearnersById, ...payload.LearnersById };
+      state.avatars = { ...state.avatars, ...payload.avatars };
+    },
+    fetchThreadFailed: (state) => {
+      state.status = RequestStatus.FAILED;
+    },
+    setSortedBy: (state, { payload }) => {
+      state.sortedBy = payload;
+      state.pages = [];
+    },
+    setStatusFilter: (state, { payload }) => {
+      state.filters.status = payload;
+      state.pages = [];
+    },
+    setAllPostsTypeFilter: (state, { payload }) => {
+      state.filters.allPosts = payload;
+      state.pages = [];
+    },
+    setMyPostsTypeFilter: (state, { payload }) => {
+      state.filters.myPosts = payload;
+      state.pages = [];
+    },
+    setSearchQuery: (state, { payload }) => {
+      state.filters.search = payload;
+      // Search doesn't work with following
+      state.filters.status = state.filters.status === PostsStatusFilter.FOLLOWING
+        ? PostsStatusFilter.ALL
+        : state.filters.status;
+      state.pages = [];
+    },
+  },
```

Comment:
```suggestion

```

Reply:
A lot of code related to threads in this file. Please clean it up.

Reply:
Code updated